viewer-prototype/src/browser/trace-viewer/trace-viewer-menu.ts

```diff
@@ -0,0 +1,70 @@
+import { injectable, inject } from 'inversify';
+import { MenuContribution, MenuModelRegistry } from '@theia/core/lib/common';
+import { CommonMenus } from '@theia/core/lib/browser';
+import URI from '@theia/core/lib/common/uri';
+import { TraceViewerWidget, TraceViewerWidgetOptions } from './trace-viewer';
+import { WidgetOpenHandler } from '@theia/core/lib/browser';
+import { Command, CommandRegistry, CommandContribution } from '@theia/core';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
+import { FileDialogService, OpenFileDialogProps } from '@theia/filesystem/lib/browser';
+
+export namespace TraceMenuCommand {
+    export const OPEN: Command = {
+        id: 'open-trace',
+        label: 'Open Trace'
```

Comment:
I suggest leaving it singular as "Open Trace". It's expected that in a command like this you can explicitly decide to open multiple traces in the same folder. For example, in the VS Code "Open File" dialogue box you can select multiple files to open (when holding Ctrl).

Having the implicit ability to open multiple traces by selecting a directory (but only for a certain trace type) isn't something I would necessarily want to advertise from a UX perspective. It shows inconsistencies in the interaction rules of the tool universe and could create distrust in a user.

Are there other actions that are implicit in this "Open" operation? Is it possible to open a Trace Compass Experiment with this?

Reply:
> I suggest leaving it singular as "Open Trace". It's expected that in a command like this you can explicitly decide to open multiple traces in the same folder. For example, in the VS Code "Open File" dialogue box you can select multiple files to open (when holding Ctrl). The "Open Folder" option also allows you to open multiple folders.
> 
Just a note, that this will require that the user knows the details of the used tracer and which traces the user needs to select and where these traces are located. In my experience, this is often not known. 
For example, for an LTTng session there is a non-tivial directory structure with kernel in the session root directory and one or more UST traces further down in the tree. Don't forget that CTF traces are directories.
The implementatoin supports opening LTTng kernel and UST together for trace correlation by selecting the session root (which is main use case of Trace Compass). 

In the future, I think will have to provide a service ("trace detection and experiment configuration service") for the user to make it easy to open a set of traces of a system that needs to be analyzed and put together. This service will have to be configurable for different tracers. It's related to #8. 

> Having the implicit ability to open multiple traces by selecting a directory (but only for a certain trace type) isn't something I would necessarily want to advertise from a UX perspective. It shows inconsistencies in the interaction rules of the tool universe and could create distrust in a user.
> 
I understand and it make sense. We don't want to break user experience, while giving the user a good experience for trace viewer.

> Are there other actions that are implicit in this "Open" operation? Is it possible to open a Trace Compass Experiment with this?
No, there is no other implicit "Open" operation. The only way to open an experiment with multiple traces is to select a directory with multiple CTF in it.

Now, we need to think about the way forward and what to include in the MVP. Opening multiple traces together is required for the MVP. I see the following 2 options (should be discussed in a different issue tracker):
Option 1)
Keep the implementation as is. Name all commands "Open Trace". Allow to select a directory and scan the sub-directories for CTF traces and open them together. 

Advantages: is that it's already supported. 
Disadvantages: It has an implicit and unexpected behavior when selecting a directory. Selecting a large directory can cause some troubles too.

Option 2)
-  From the context sensitive-menu of file explorer: Support opening a single trace only.
-  From new open button or CTRL+SHIFT+P: Allow user to select multiple directories and pass them to the open trace command. In the open trace command we post each trace to the trace server and then post the experiment containing the traces which will be shown in the trace explorer.

Advantages: Explicit opening of multiple traces in a single experiment. 
Disadvantages: More involving from implementation point of view (might not be ready for the MVP). User has to know about which traces to select.

Reply:
> This will require that the user knows the details of the used tracer and which traces the user needs to select and where these traces are located. In my experience, this is often not known.

For the MVP I'm still applying the assumption that users have the perfect theoretical background for using the tool. I agree that this solution for opening data sources does not represent the complex use cases and guidance required in the final tool. 

> Now, we need to think about the way forward and what to include in the MVP. Opening multiple traces together is required for the MVP.

I'll respond directly in the new issue #189.